include/aspect/material_model/interface.h

```diff
@@ -674,6 +674,41 @@ namespace aspect
 
 
     /**
+     * Additional output fields for reaction rates to be added to
+     * the MaterialModel::MaterialModelOutputs structure and filled in the
+     * MaterialModel::Interface::evaluate() function.
+     *
+     * These reaction rates are only used in the operator_splitting nonlinear
+     * solver scheme, which allows it to solve reactions between compositional
+     * fields decoupled from the advection, and using a different time step size.
+     * In this case, they are used in addition to (and independent from) any
+     * reaction_terms that a material model defines, which are assembled as usual.
+     * For any other solver scheme, these values are ignored.
+     *
+     * In contrast to the reaction_terms, which are actual changes in composition
+     * rather than reaction rates, and assume equilibrium between the compositional
+     * fields, the reacion_rates defined here allow for reaction processes that are
+     * on shorter time scales than the advection, and disequilibrium reactions.
+     */
+    template <int dim>
+    class ReactionRateOutputs : public NamedAdditionalMaterialOutputs<dim>
+    {
+      public:
+        ReactionRateOutputs (const unsigned int n_points,
+                             const unsigned int n_comp);
+
+        virtual std::vector<double> get_nth_output(const unsigned int idx) const;
+
+        /**
+         * Reaction rates at the evaluation points passed to
+         * the instance of MaterialModel::Interface::evaluate() that fills
+         * the current object.
+         */
+        std::vector<std::vector<double> > reaction_rates;
```

Comment:
The reaction_terms don't have the name "composition" in them either, and I think it would make sense to have similar names for those two things. I guess to me "reaction" already implied that this is for a compositional field. 

> Also: if I have a reaction that turns a composition into heat, I need to make a separate heating plugin, right?

Correct. 

> Wouldn't it make more sense if there was a temperature reaction rate in here too? Or is this not a typical case?

Hm, we could move the rates_of_temperature_change from the heating model to the material model, but it is only used in the temperature equation and nowhere else, so maybe that would be a bit confusing. On the other hand, we also have the entropy derivatives in the material model, and it's the same for them. I guess my line of thought was that a "rate of temperature change" is something more abstract than a material property, it is closer to a "heating source term" than to most of the other material model outputs, which specifically relate to a specific property of the material.  
The only case I can think of where you would want to relate changes in composition and temperature are latent heat affect, and we already have a heating model for that ("latent heat melt" already works with operator splitting). There might be other uses, but nothing that I can think of. So I don't think there would be many cases where one would have to create a new heating model. 
The advantage of splitting it is also that you can mix and match, you can use the "latent heat melt" heating model (which is quite general) with different melting models, and don't have to copy all of that stuff. And as we noticed that more and more material models are basically just an existing model with some small things changed, I think it makes a lot of sense to keep these things separate and more modular.